truck/src/main/java/com/example/demo/truck/TruckService.java

```diff
@@ -22,37 +22,25 @@ public Truck getTruck(int idTruck) {
 
     public Truck addTruck(Truck newTruck) {
         if (truckGoingToPosition(newTruck.getId_position())) {
-            if (truckRepository.existsById(newTruck.getId_truck())) {
-                throw new IllegalArgumentException("Id " + newTruck.getId_truck() + " déja utilisé");
-            } else {
-                return truckRepository.save(newTruck);
-            }
+            throw new IllegalArgumentException("Id " + newTruck.getId_truck() + " déja utilisé");
```

Comment:
When is the truck saved
```suggestion
            throw new IllegalArgumentException("Id " + newTruck.getId_truck() + " déja utilisé");
```

Reply:
Fixed